src/services/assetService.ts

```diff
@@ -192,8 +192,8 @@ export class AssetService {
             if (!types) {
                 console.error(interpolate(strings.editorPage.assetWarning.incorrectFileExtension.failedToFetch, { fileName: corruptFileName.toLocaleUpperCase() }));
             }
-            // If file was renamed/spoofed - fix file extension to true MIME type and show message
-            else if (!types.includes(assetFormat)) {
+            // If file was renamed/spoofed - fix file extension to true MIME if it's type is in supported file types and show message
+            else if (types[0] && !types.includes(assetFormat)) {
                 assetFormat = types[0];
                 console.error(`${strings.editorPage.assetWarning.incorrectFileExtension.attention} ${corruptFileName.toLocaleUpperCase()} ${strings.editorPage.assetWarning.incorrectFileExtension.text} ${corruptFileName.toLocaleUpperCase()}`);
```

Comment:
If my understanding is correct, the error should be logged whenever types.includes(assetFormat) == false, while assetFormat overwritten should only happen if types is not empty.

Reply:
yes